server/test/auth.spec.js

```diff
@@ -0,0 +1,141 @@
+        form: {
+          email: "test@example.com",
+          name: "",
+          password: "password",
+          confirmPassword: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `blank password returns "${response[0].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          password: "",
+          confirmPassword: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `blank confirmPassword returns "${response[0].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          password: "password",
+          confirmPassword: "",
+        },
+        response: response[0],
+      },
+      {
+        name: `missing name returns "${response[0].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          password: "password",
+          confirmPassword: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `missing email returns "${response[0].join("--")}"`,
+        form: {
+          name: "name",
+          password: "password",
+          confirmPassword: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `missing password returns "${response[0].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          confirmPassword: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `missing confirmPassword returns "${response[0].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          password: "password",
+        },
+        response: response[0],
+      },
+      {
+        name: `short password returns "${response[2].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          password: "pass",
+          confirmPassword: "pass",
+        },
+        response: response[2],
+      },
+      {
+        name: `password mismatch returns "${response[3].join("--")}"`,
+        form: {
+          email: "test@example.com",
+          name: "name",
+          password: "password",
+          confirmPassword: "passwords",
+        },
+        response: response[3],
+      },
+      {
+        name: `existing email returns "${response[1].join("--")}"`,
+        form: {
+          email: "user@example.com",
+          name: "Not User One",
+          password: "password",
+          confirmPassword: "password",
+        },
+        response: response[1],
+      },
+    ];
+    tests.forEach(({ name, form, response: [message, status] }) => {
+      it(name, (done) => {
+        chai
+          .request(app)
+          .post(`/api/register`)
+          .type("form")
+          .send(form)
+          .end((err, res) => {
+            res.text.should.eql(message);
+            res.should.have.status(status);
```

Comment:
I'm assuming with these tests we are using the dev database taken from process.env.DB_URL. I'd prefer that we actually used something that mocked our database so we actually don't even need to worry about tests conflicting with our databases - something like https://github.com/nodkz/mongodb-memory-server

Reply:
Does this create a temporary instance in Mongo with the same version of the server? I was worried about version control, but if this can run on the same instance in a sandbox then I'll add it. (Also, I definitely need to add a db teardown function to run after every suite.)

Reply:
You can configure it - https://github.com/nodkz/mongodb-memory-server#configuring-which-mongod-binary-to-use